libs/liblobby/lobby/interface.lua

```diff
@@ -648,7 +648,128 @@ Interface.commands["FRIENDREQUESTLISTEND"] = Interface._OnFriendRequestListEnd
 -- Battle commands
 ------------------------
 
+
+-- if num1 > num2 return 1
+-- if num1 < num2 return 0
+-- if num1 = num2 return -1
+local function bignumStr_isGreater(num1, num2)
+	if #num1 > #num2 then
+		return 1
+	end
+	if #num1 < #num2 then
+		return 0
+	end
+  
+	local dig1, dig2
```

Comment:
you don't need to compare digit by digit, you can just do `num1 < num2` because once you know the length is the same, you can just do lexicographical comparison on strings as numbers are sequentially in ascii.

So you end up with something like:
```
local function bignum_compare(num1, num2)
	if #num1 < #num2 then
		return -1
	elseif #num1 > #num2 then
		return 1
	elseif num1 == num2 then
		return 0
	elseif num1 < num2 then
		return -1
	else
		return 1
	end
end
```